sys/crypto/psa_riot_cipher/chacha20.c

```diff
@@ -65,22 +66,32 @@ psa_status_t psa_cipher_chacha20_set_iv(psa_cipher_chacha20_ctx_t *ctx,
     DEBUG("RIOT ChaCha20 Cipher set IV");
     psa_status_t status = PSA_ERROR_CORRUPTION_DETECTED;
 
```

Comment:
I guess you should add a check to make sure that IV hasn't been set before. According to the [PSA API](https://arm-software.github.io/psa-api/crypto/1.2/api/ops/ciphers.html#c.psa_cipher_set_iv), you should return `PSA_ERROR_BAD_STATE` in that case. Same goes for the HW backend, or you move the check one up to `psa_crypto.c`.

Reply:
That is already checked in psa_cipher_set_iv() in psa_crypto.c

Reply:
Okay, perfect! (I was too lazy to double-check)